sherpa/astro/data.py

```diff
@@ -4807,6 +4865,10 @@ def notice(self, lo=None, hi=None, ignore=False, bkg_id=None):
         #
         for bid in bkg_ids:
             bkg = self.get_background(bid)
+            if bkg is None:
```

Comment:
Maybe this is OK, but it look odd to me. `bkg_ids` is returned from `self.background_ids` and I would expect every background to have an identifier. Or is there a situation where that's not the case?`

Reply:
Or is this for the case when no background it set at all? But then the comment in the line below is confusing.

Reply:
So this is a case where I claim our API doesn't help us. We should have a way to be able to loop through the backgrounds, which would mean we don't need the `if bkg is None` statement to please the type checker

At the moment, since `get_background` returns `DataPHA | None` . the type checker has no way to know that `get_background` can not return `None` here.

I'm not sure the best way to square the circle

- live with the old code, and annoy the type checker gods
- annoy readers with pointless code
- address the API issues, which is just not a good thing to be doing at this stage of the release cycle (well, not when I have other changes I want to get in ;-)

Reply:
For some reason, I've never managed to get a type checker to work with my editor. I know many people do and it should work, but something is off and I never bothered to spend the time to track it down. So I'm fine with annoying the type checker gods ... but you do much more work in the codebase than I do and the typecheckers did help us (more precisely: you) find a few useful things in the past. So, if you prefer "pointless code" that's also fine with me - just click "resolve the conversation"